### ISSUE TITLE
AutoMapper 7.0.1 PDB for .NET Framework 4.5 is portable PDB and cannot be indexed

### ISSUE THREAD

**Issue by lupestro** [AutoMapper/AutoMapper#2730]
### Expected behavior

When a TFS build gets to the indexing step, it can successfully index all PDBs either generated by the build or received from NuGet. 

### Actual behavior

As the TFS build is nearly completed, it gets to the source indexing stage.
```
Get Impacted Tests, Index Sources and Publish Symbols
-------------------------------------------------------------
Indexed source information could not be retrieved from C:\Builds\2\******\******\Binaries\AutoMapper.pdb. Error: Symbol indexes could not be retrieved.
```
This causes the build to go yellow (I didn't even know the TFS build_had_ a yellow! 🙂 ) and while I could perhaps successfully deploy it this way - I'm about to see if our uDeploy script permits it - our site folks won't be ecstatically happy unless I can make the build "run green".

### Steps to reproduce

Any attempt to index this PDB within any .NET 4.5 project in a TFS build. 

### Discussion

The root cause is a known problem: this is a portable PDB, and the indexer in the TFS build can't index portable PDBs. I know that this is a portable PDB because when I analyze it with dotPeek, I see it clearly in the data they return.

Our site's build infrastructure is very old, and our build targets are still .NET Framework 4.5.1 and C#5. It would be very useful for us if the PDBs in the /net45 area of AutoMapper were full PDBs rather than portable PDBs so they can be used with the build tools that were supplied in the .NET 4.5 era. 

In the meantime, assuming this takes a little while to adjust, I'll have to back off my AutoMapper to an earlier version. AutoMapper 6.1.1, which we were using before Friday, didn't supply PDBs, so we didn't encounter the problem. I'll have to see if 6.2.2, the last version before 7.0, also omits them.

**Comment by lupestro** [comment]
I backed off to using 6.2.2 so I could deliver our stuff. I delivered this afternoon. That was my solution at our end. We no longer face a project crisis from this. However, we are no longer keeping up with the version flow of one of our dependencies, and that's a potential risk I'd prefer to mitigate. 

I could look at doing a PR for appropriate PDBs per platform, but I only became aware that the PDB format was undergoing change as I dug into our build problem this morning and I don't feel like I have a complete handle on the situation yet. After only a couple of hours of research, I think I know: 

* We always want portable PDBs  for .NET Core. 
* We always want traditional "big" PDBs for older .NET Framework versions.
* Is there a recent version of .NET Framework and the tooling for which portable PDBs work for debugging? Can anyone confirm from personal experience?
* Is there a recent version of .NET Framework and the tooling where indexing will automatically either index or skip indexing of portable PDBs it finds without treating this as a yellow-flag build issue?
* If there is such a version, we should use portable PDBs for that version of .NET Framework forward and deliver portable PDBs for .NET standard builds.
* If there is no such version, we should use "big" PDBs for *all* .NET Framework versions. For .Net Standard, we'd probably be stuck with no PDB at all, since whatever we did would be "wrong" for someone, although I suppose portable PDBs could be delivered, especially if debugging worked for them but indexing didn't.

Does the above reasoning sound correct? If anybody wants to offer me some guidelines or point me to further salient information, I suppose I could give it a shot. Or I could just address the narrow issue and make sure net45 delivers traditional PDBs.
